jsonpickle/ext/pandas.py

```diff
@@ -120,27 +116,102 @@ def restore(self, data):
 class PandasIndexHandler(BaseHandler):
     pp = PandasProcessor()
 
+    index_constructor = pd.Index
+    name_bundler = lambda _, obj: {'name': obj.name}
+
     def flatten(self, obj, data):
-        meta = {'dtype': str(obj.dtype), 'name': obj.name}
-        buf = json.dumps(obj.tolist())
+        name_bundle = self.name_bundler(obj)
+        meta = dict(dtype= str(obj.dtype), **name_bundle)
+        buf = encode(obj.tolist())
         data = self.pp.flatten_pandas(buf, data, meta)
         return data
 
     def restore(self, data):
         buf, meta = self.pp.restore_pandas(data)
         dtype = meta.get('dtype', None)
-        name = meta.get('name', None)
-        idx = pd.Index(json.loads(buf), dtype=dtype, name=name)
+        name_bundle = {k: v for k, v in meta.items() if k in {'name', 'names'}}
+        idx = self.index_constructor(decode(buf), dtype=dtype, **name_bundle)
         return idx
 
 
+class PandasPeriodIndexHandler(PandasIndexHandler):
+    index_constructor = pd.PeriodIndex
+
+
+class PandasMultiIndexHandler(PandasIndexHandler):
+    name_bundler = lambda _, obj: {'names': obj.names}
+
+
+class PandasTimestampHandler(BaseHandler):
+    pp = PandasProcessor()
+
+    def flatten(self, obj, data):
+        meta = {"isoformat": obj.isoformat()}
+        buf = ""
+        data = self.pp.flatten_pandas(buf, data, meta)
+        return data
+
+    def restore(self, data):
+        _, meta = self.pp.restore_pandas(data)
+        isoformat = meta['isoformat']
+        obj = pd.Timestamp(isoformat)
+        return obj
+
+
+class PandasPeriodHandler(BaseHandler):
+    pp = PandasProcessor()
+
+    def flatten(self, obj, data):
+        meta = {"start_time": encode(obj.start_time), "freqstr": obj.freqstr}
+        buf = ""
+        data = self.pp.flatten_pandas(buf, data, meta)
+        return data
+
+    def restore(self, data):
+        _, meta = self.pp.restore_pandas(data)
+        start_time = decode(meta['start_time'])
+        freqstr = meta['freqstr']
+        obj = pd.Period(start_time, freqstr)
+        return obj
+
+
+class PandasIntervalHandler(BaseHandler):
+    pp = PandasProcessor()
+
+    def flatten(self, obj, data):
+        meta = {"left": encode(obj.left), "right": encode(obj.right), "closed": obj.closed}
+        buf = ""
+        data = self.pp.flatten_pandas(buf, data, meta)
+        return data
+
+    def restore(self, data):
+        _, meta = self.pp.restore_pandas(data)
+        left = decode(meta['left'])
+        right = decode(meta['right'])
+        closed = str(meta['closed'])
+        obj = pd.Interval(left, right, closed=closed)
+        return obj
+
+
 def register_handlers():
+    register_numpy_handlers()
```

Comment:
This seems not ideal, but I was running into some strange errors when running the tests vs python 2 that this fixed. It seems related to [this](https://github.com/jsonpickle/jsonpickle/issues/239) since the errors only occurred in python 2 and not python 3.